vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -0,0 +1,484 @@
+	CVEsForKnownRepos   int
+	OSVRecordsGenerated int
+	Outcomes            map[cves.CVEID]ConversionOutcome // Per-CVE-ID record of conversion result.
+}
+
+// References with these tags have been found to contain completely unrelated
+// repositories and can be misleading as to the software's true repository,
+// Currently not used for this purpose due to undesired false positives
+// reducing the number of valid records successfully converted.
+var RefTagDenyList = []string{
+	// "Exploit",
+	// "Third Party Advisory",
+	"Broken Link", // Actively ignore these though.
+}
+
+// VendorProducts known not to be Open Source software and causing
+// cross-contamination of repo derivation between CVEs.
+var VendorProductDenyList = []cves.VendorProduct{
+	// Three strikes and the entire netapp vendor is out...
+	{Vendor: "netapp", Product: ""},
+	// [CVE-2021-28957]: Incorrectly associates with github.com/lxml/lxml
+	{Vendor: "oracle", Product: "zfs_storage_appliance_kit"},
+	{Vendor: "gradle", Product: "enterprise"}, // The OSS repo gets mis-attributed via CVE-2020-15767
+}
+
+// formatDateTimeRFC3339 parses an ISO-like date string and returns RFC3339 format or original on error.
+func formatDateTimeRFC3339(dateStr string) string {
+	if dateStr == "" {
+		return ""
+	}
+	// Try parsing common ISO8601 variations
+	layouts := []string{
+		time.RFC3339Nano,           // "2006-01-02T15:04:05.999999999Z07:00"
+		time.RFC3339,               // "2006-01-02T15:04:05Z07:00"
+		"2006-01-02T15:04:05Z",     // ISO 8601 UTC
+		"2006-01-02T15:04:05.000Z", // Explicit milliseconds UTC
+	}
+	var t time.Time
+	var err error
+	for _, layout := range layouts {
+		t, err = time.Parse(layout, dateStr)
+		if err == nil {
+			break // Parsed successfully
+		}
+	}
+
+	if err != nil {
+		fmt.Printf("Warning: Could not parse date string '%s': %v. Returning original.\n", dateStr, err)
+		return dateStr // Return original if parsing fails
+	}
+	return t.UTC().Format(time.RFC3339) // Format in UTC RFC3339
+}
+
+// BORROWED FROM VULNS.GO
+// FromCVE creates a minimal OSV object from a given CVEItem and id.
+// Leaves affected and version fields empty to be filled in later with AddPkgInfo
+func FromCVE(id cves.CVEID, cve cves.CVE5) (*vulns.Vulnerability, []string) {
+	aliases, related := vulns.ExtractReferencedVulns(id, cves.CVEID(cve.Metadata.CVEID), cve.Containers.CNA.References)
+	v := vulns.Vulnerability{
+		ID:      string(id),
+		Details: cves.EnglishDescription(cve.Containers.CNA.Descriptions),
+		Aliases: aliases,
+		Related: related,
+	}
+	var notes []string
+
+	v.Published = formatDateTimeRFC3339(cve.Metadata.DatePublished)
+	v.Modified = formatDateTimeRFC3339(cve.Metadata.DateUpdated)
+	v.References = vulns.ClassifyReferences(cve.Containers.CNA.References)
+	// v.AddSeverity(cve.Metrics)
+	return &v, notes
+}
+func ExtractVersionInfo(cve cves.CVE5, refs []string, validVersions []string, httpClient *http.Client) (v models.VersionInfo, notes []string) {
```

Comment:
This function is quite long and doing multiple things. Can we break down into some smaller helper functions to improve readabilty?